irods_testing_environment/odbc_setup.py

```diff
@@ -273,7 +275,10 @@ def configure_odbc_driver_mysql_80(csp_container, odbc_driver):
     """
     if not odbc_driver:
         odbc_driver = download_mysql_odbc_driver(
-            'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.0/mysql-connector-odbc-8.0.33-linux-glibc2.28-x86-64bit.tar.gz')
+            'https://dev.mysql.com/get/Downloads/Connector-ODBC/9.1/mysql-connector-odbc-9.1.0-linux-glibc2.28-x86-64bit.tar.gz') # THIS WORKS!
+            #'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.4/mysql-connector-odbc-8.4.0-linux-glibc2.28-x86-64bit.tar.gz') # THIS WORKS!
+            #'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.0/mysql-connector-odbc-8.0.40-linux-glibc2.28-x86-64bit.tar.gz') # DOES NOT WORK FOR SOME REASON
+            #'https://dev.mysql.com/get/Downloads/Connector-ODBC/8.0/mysql-connector-odbc-8.0.33-linux-glibc2.28-x86-64bit.tar.gz')
```

Comment:
Which one is most appropriate?

What I can say is that dealing with MySQL ODBC drivers is a real pain. The drivers don't seem to provide any details as to why one version works over another and finding documentation on that isn't obvious.

Regardless, I've seen the tests pass with v9.1.0 and I've seen tests pass with v8.4.0. v8.4.0 is used for MariaDB testing so using v9.1.0 allows us to confirm compatibility with another driver, even if the database is different (i.e. MySQL 8 vs MariaDB).

I'm leaning towards using v9.1.0 for MySQL 8.

Thoughts?

Reply:
@SwooshyCueb did a lot of work in this area. Almost certainly has better input than me. :)

According to their website, 9.1.0 is officially compatible with MySQL 8.0: https://dev.mysql.com/doc/connector-odbc/en/connector-odbc-versions.html We allow you to specify an alternative ODBC driver through options in the testing environment, so I don't see a problem with using the latest.

Reply:
Are you saying we expose an option via the command line that let's you use a different ODBC driver?

Reply:
Many of our scripts have the `--odbc-driver-path` option which allows you to point at an archive file on the host to slurp and install in the catalog provider instance. It's not "smart", but it does allow you to use different ODBC drivers should you need that.

Reply:
Didn't know that, although I'm sure I've seen that option many times while looking at the help text lol.

Reply:
Moving forward with the MySQL 9.1 ODBC driver.

I'll open an issue for MariaDB, so we don't forget about it.

Reply:
Created #237